src/main/java/io/jenkins/plugins/jfrog/CliEnvConfigurator.java

```diff
@@ -79,4 +79,11 @@ private static void excludeProxyEnvFromPublishing(EnvVars env) {
         String jfrogCliEnvExclude = env.getOrDefault(JFROG_CLI_ENV_EXCLUDE, JFROG_CLI_DEFAULT_EXCLUSIONS);
         env.put(JFROG_CLI_ENV_EXCLUDE, String.join(";", jfrogCliEnvExclude, HTTP_PROXY_ENV, HTTPS_PROXY_ENV));
     }
+
+    static String noProxyExtractor(String noProxyList) {
+        // Trim leading and trailing spaces, Replace '|' with spaces and normalize whitespace
+        String noProxyListRemoveSpaceAndPipe = noProxyList.trim().replaceAll("[\\s|]+", ";");
+        // Replace multiple semicolon with a single semicolon, and remove the last one if present
+        return noProxyListRemoveSpaceAndPipe.replaceAll(";+", ";").replaceAll(";$", "");
```

Comment:
A dangling `;` character can also be in the beginning, not only in the end.
```suggestion
        return noProxyListRemoveSpaceAndPipe.replaceAll(";+", ";").replaceAll("^;|;$", "");
```